src/saving/SaveList.cs

```diff
@@ -53,12 +58,17 @@ public override void _Ready()
 
     public override void _Process(float delta)
     {
-        if (AutoRefreshOnFirstVisible && !refreshedAtLeastOnce && IsVisibleInTree())
+        if ((AutoRefreshOnFirstVisible && !refreshedAtLeastOnce && IsVisibleInTree()) ||
+            (AutoRefreshOnBecomingVisible && !wasVisible && IsVisibleInTree()))
         {
             Refresh();
+            wasVisible = true;
             return;
         }
 
+        if (!IsVisibleInTree())
```

Comment:
Then this should use `isCurrentlyVisible` so that that method is called only once.